### PR TITLE
fix custom op version compatibility

### PR DESCRIPTION
**Pull request by zhyncs**
## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html).

### CODE REVIEW

File: python/sglang/srt/layers/rotary_embedding.py [hunk 1]
```diff
@@ -51,7 +53,7 @@ def _apply_rotary_emb(
         return torch.stack((o1, o2), dim=-1).flatten(-2)
 
 
-@CustomOp.register("rotary_embedding")
+@register_custom_op("sglang_rotary_embedding")
```
Comment (zhyncs) [inline python/sglang/srt/layers/rotary_embedding.py:56]: ref https://github.com/sgl-project/sglang/blob/a69cb5cff7389fb6ce1b4c45c52b6796e78ce0f3/python/pyproject.toml#L30

File: python/sglang/srt/models/minicpmv.py [hunk 2]
```diff
@@ -39,10 +39,14 @@
 from torch import nn
 from torch.nn.init import trunc_normal_
 from transformers import PretrainedConfig
-from vllm.model_executor.layers.resampler import get_2d_sincos_pos_embed
-from vllm.model_executor.layers.sampler import SamplerOutput, get_sampler
-from vllm.model_executor.models.module_mapping import MultiModelKeys
-from vllm.model_executor.sampling_metadata import SamplingMetadata
+
+try:
```
Comment (zhyncs) [inline python/sglang/srt/models/minicpmv.py]: ref https://github.com/sgl-project/sglang/blob/a69cb5cff7389fb6ce1b4c45c52b6796e78ce0f3/python/pyproject.toml#L30

File: python/sglang/srt/torch_memory_saver_adapter.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
 from contextlib import contextmanager
 
-try:
-    import torch_memory_saver
+import torch
 
-    _primary_memory_saver = torch_memory_saver.TorchMemorySaver()
-except ImportError:
-    pass
+if hasattr(torch.cuda, "MemPool"):
```
Comment (zhyncs) [inline python/sglang/srt/torch_memory_saver_adapter.py]: ref https://github.com/sgl-project/sglang/blob/a69cb5cff7389fb6ce1b4c45c52b6796e78ce0f3/python/pyproject.toml#L30

**Review by zhyncs — COMMENTED** [review]
vllm==0.6.3.post1,torch==2.4.0

vllm==0.6.4.post1,torch==2.5.1

Our previous CI environment was torch 2.5.1, so the issues didn't surface, in fact, there were a bunch of problems with torch 2.4.0.

File: python/pyproject.toml [hunk 4]
```diff
@@ -27,7 +27,7 @@ runtime_common = [
 ]
 srt = [
     "sglang[runtime_common]", "cuda-python",
-    "sgl-kernel>=0.0.2.post14", "torch", "vllm>=0.6.3.post1,<=0.6.4.post1",
+    "sgl-kernel>=0.0.2.post14", "torch", "vllm==0.6.4.post1",
```
Comment (zhyncs) [inline python/pyproject.toml:30]: As we discussed offline, upgrading torch to 2.5.1 is fine. I've changed vLLM to 0.6.4.post1 because the main branch currently has issues with torch 2.4.0, and we can't upgrade vLLM beyond v0.6.5 due to conflicts with `outlines`. cc @merrymercy
